Automattic/wp-calypso
# Store: Fix inventory behavior when inventory management has been turned off

Fixes #24410.

This PR fixes the behavior reported at p90Yrv-El-p2, where the inventory controls are still available in Calypso if inventory management has been disabled in wp-admin. It grays out the input box and provides a link for reenabling it in wp-admin (since we don't expose those settings anywhere in the Calypso interface).

<img width="816" alt="screen shot 2018-04-24 at 1 25 17 pm" src="https://user-images.githubusercontent.com/689165/39203496-f6c14086-47c2-11e8-8668-8b2091341f79.png">

<img width="691" alt="screen shot 2018-04-24 at 1 25 09 pm" src="https://user-images.githubusercontent.com/689165/39203506-fa5af836-47c2-11e8-89a5-b253858596c1.png">

To Test:
* Go to `/wp-admin/admin.php?page=wc-settings&tab=products&section=inventory` and disable inventory/stock management. 
* Go to Calypso Store and create a new product. Verify that you see the message above.
* Switch the product type to a variable product and create some variations. Verify you see the message above.
* Test in the edit context.
* Re-enable the setting, and verify you can still update inventory settings.
